blender/LilySurfaceScraper/Scrapers/AbstractScraper.py

```diff
@@ -166,12 +176,73 @@ def clearString(self, s):
         return ''.join(filter(lambda x: x in printable, s))
 
     def fetchVariantList(self, url):
+        # **must have self.asset_name**
+
+        # get asset name and variants
+        variants = self._fetchVariantList(url)
+        assetName = self.asset_name
+
+        root = self.getTextureDirectory(os.path.join(self.home_dir, assetName))
+
+        # download thumbnail and make metadata file if meta file is not present
+        metadataFile = os.path.join(root, ".meta")
+        if not os.path.isfile(metadataFile):
+            thumbnailName = self.downloadThumbnail(root, assetName)
+
+            metadata = {
+                "name": assetName,
+                "scraper": self.__class__.__name__,
+                "fetchUrl": url,
+                "thumbnail": thumbnailName,
+                "variants": variants,
+            }
+            with open(metadataFile, "w") as f:
+                json.dump(metadata, f, indent=4)
+        return variants
+
+    def downloadThumbnail(self, assetPath, assetName):
+        thumbnailUrl = self.getThumbnail(assetName)
+        ext = None
+        if thumbnailUrl is None:
+            print("no thumbnail found, not downloading")
+        else:
+            thumbnailReq = self._fetch(thumbnailUrl)
+            if thumbnailReq is None:
+                return None
+            thumbnailType = thumbnailReq.headers["Content-Type"]
+            if thumbnailType == 'image/png':
+                ext = "png"
+            elif thumbnailType == "image/jpeg":
+                ext = "jpg"
+            else:
+                print(f"thumbnail type '{thumbnailType}' is not a valid type")
+
+        if ext is None:
+            return None
+        thumbnailName = f"thumb.{ext}"
+        with open(os.path.join(assetPath, thumbnailName), "wb") as f:
+            f.write(thumbnailReq.content)
+        return thumbnailName
+
+    def _fetchVariantList(self, url):
         """Get a list of available variants.
         The list may be empty, and must be None in case of error."""
         raise NotImplementedError
-    
+
     def fetchVariant(self, variant_index, material_data):
         """Fill material_data with data from the selected variant.
         Must fill material_data.name and material_data.maps.
         Return a boolean status, and fill self.error to add error messages."""
         raise NotImplementedError
+
+    def getThumbnail(self, assetName):
```

Comment:
Preferably use snake_case for variable names, as advised by Python standard, and more importantly to ensure consistency across the code base of this repo.